test/WebSites/FormatFilterWebSite/Controllers/FormatFilterController.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc;
+
+namespace FormatFilterWebSite
+{
+    [FormatFilter]
+    public class FormatFilterController : Controller
+    {   
+        public Product GetProduct(int id)
+        {
+            return new Product() { SampleInt = id };
+        }
+
+        [Produces("application/custom", "application/json", "text/json")]
+        public Product ProducesMethod(int id)
+        {
+            return new Product() { SampleInt = id };
+        }
+    }
```

Comment:
@rynowak -- @yishaigalatzer wanted more specific tests than a wildcard produces filter. So I have added tests to this file: test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs. The tests are FormatFilter_MoreSpecificThan_Produces and FormatFilter_LessSpecificThan_Produces. I have added another one with wildcard produces.